src/hooks/index.ts

```diff
@@ -1,6 +1,6 @@
 import { useLocation } from 'react-router';
 
-const useQuery = (): any => {
+const useQuery = () => {
   return new URLSearchParams(useLocation().search);
 };
```

Comment:
`query string` 은 받을 수 있지만 `query params` 는 못가져오는데 `query` 를 하나의 훅스로 관리하는건 어떨까요?



```suggestion
import queryString from 'query-string'

const useQuery = () => {
  const params = useParams()
  return {
    query: {
      ...queryString.parse(location.search),
      ...params,
    }
  }
}
```
이렇게 작성 해볼 수 있을거 같아요!
> `query-string` 라이브러리 사용하시면 쿼리스트링을 키밸류 오브젝트로 관리해줘서 굉장히 편해요!

Reply:
해당 리뷰를 반영하는 도중에 질문이 있습니다!  
`useQueryStringAndParam`이름으로 hook을 만들고 `Auth.tsx`파일에서 `const { authorizationToken } = useQueryStringAndParam().query;`로 코드를 작성하면`Property 'authorizationToken' does not exist on type '{}'`라는 에러가 뜨는데욤.. 해서 아래처럼 리턴타입을 `any`로 지정해주었는데 괜찮을까요..? 실제 코드가 돌기 전까지는 어떤 key&value가 들어올지 몰라 타입 지정하기가 애매해서 `any`를 지정해주었습니다. 아니면 어떤 key&value가 들어올지 확신(?)을 가지고 결정 될 때마다 타입을 미리미리 추가해주어야하는 것인지... 제가 TS에 익숙하지않아서 기초적인 질문인것같지만😥😥  이런 경우에는 어떻게 타입을 지정하시는지 궁금해서 여쭈어봅니다!
```
  return useMemo((): any => {
    return {
      query: {
        ...queryString.parse(window.location.search),
        ...params,
      },
    };
  }, [params]);
```

Reply:
`query`의 타입이 따로 타입을 주지 않으면 any 로 잡히지 않나요?

Reply:
`query`의 타입이 `{}`요걸로 잡혀있습니당 아니면 `any` 대신에 아래처럼 타입을 만들어서 넣어주는 것이 좋을까요?
```
interface QueryStringAndParam {
  query: any
}
//...생략
  return useMemo((): QueryStringAndParam => {
    return {
      query: {
        ...queryString.parse(window.location.search),
        ...params,
      },
    };
  }, [params]);
```

Reply:
`query-string` 타입을 보니 제너릭을 안받아서 애매한 상황이긴 하네요..허헛.. 조금 꼼수를 부려보자면
```ts
const useQuery = <T>() => {
  const params = useParams<T>()
  ...,
  ...,
}
```
이렇게 주시면 `query`의 타입이 `any`로 추론 될거라 아래처럼 사용할 수 있어요!
`const { authorizationToken } = useQueryStringAndParam<{authorizationToken:string}>().query`

Reply:
썩 좋은 방법은 아니라서 찜찜하긴하네요..ㅠㅠ 프론트채널에서 얘기 하거나 리팩토링 할 부분 모아서 프짞타 해도 좋을거 같아요!